catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/watcher/FilesWatcher.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.monitor.watcher;
+
+import java.io.File;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.codice.ddf.platform.util.StandardThreadFactoryBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Periodically calls {@link FileWatcher#check()} on all watched {@link FileWatcher}s. */
+public class FilesWatcher {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FilesWatcher.class);
+
+  private static final long DEFAULT_POLL_TIME_SEC = 5;
+
+  private static final String CDM_FILE_CHECK_PERIOD_PROPERTY = "org.codice.ddf.cdm.fileCheckPeriod";
+
+  private ScheduledExecutorService executorService;
+
+  private final Map<File, FileWatcher> watchers;
+
+  private long executorPollTime;
+
+  /**
+   * Creates a new FileWatcher with the system property {@link #CDM_FILE_CHECK_PERIOD_PROPERTY} poll
```

Comment:
✏️ We may want to replace `CDM_FILE_CHECK_PERIOD_PROPERTY` with the actual value, or make `CDM_FILE_CHECK_PERIOD_PROPERTY` `public` and use `{@link #CDM_FILE_CHECK_PERIOD_PROPERTY}`.